components/scream/src/share/atm_process/atmosphere_process.cpp

```diff
@@ -185,31 +185,35 @@ void AtmosphereProcess::run_precondition_checks () const {
   for (const auto& it : m_precondition_checks) {
     const auto& pc = it.second;
 
-    auto check_result = pc->check();
-    if (check_result.pass) {
+    auto res_and_msg = pc->check();
```

Comment:
You likely know this, but C++17 has a neat way to handle multiple return values. (It also fixes some `constexpr`-related defects in C++14, which has pushed us to adopt it in our EAGLES work, which is why I mention this).

Reply:
Yes, I think it's called "structured binding", or something like that. I think we officially require only C++14 though, so we can't use it (yet).

Reply:
Fair enough. But I think we should consider moving to C++17 when practical. It's a measurable improvement over C++14 (and I say this as a C++ skeptic!).

Reply:
I think next Kokkos release will be a good time to do it. IIRC, Kokkos 3.7 will _require_ C++17. The efforts to support HIP in ekat/scream might push us to adopt latest kokkos anyways.